src/liballoc/vec.rs

```diff
@@ -2390,6 +2390,21 @@ impl<T: Clone> From<&mut [T]> for Vec<T> {
     }
 }
 
+#[stable(feature = "vec_from_array", since = "1.42.0")]
+impl<T, const N: usize> From<[T; N]> for Vec<T>
+where
+    [T; N]: LengthAtMost32,
+{
+    #[cfg(not(test))]
+    fn from(s: [T; N]) -> Vec<T> {
+        <[T]>::into_vec(box s)
+    }
+    #[cfg(test)]
+    fn from(s: [T; N]) -> Vec<T> {
+        crate::slice::into_vec(box s)
```

Comment:
Shouldn't this second version also work in the `cfg(not(test))` case?

Reply:
This does not work with `cfg(not(test))`.

```
error[E0425]: cannot find function `into_vec` in module `crate::slice`
    --> src/liballoc/vec.rs:2399:23
     |
2399 |         crate::slice::into_vec(box s)
     |                       ^^^^^^^^ not found in `crate::slice`
     |
help: possible candidate is found in another module, you can import it into scope
     |
59   | use crate::slice::hack::into_vec;
```

Reply:
Possible explanation here: https://github.com/rust-lang/rust/blob/828305d10143172e0418d5d0a7111822149b4919/src/liballoc/vec.rs#L1851

Reply:
And [here is said `hack` module](https://github.com/rust-lang/rust/blob/828305d10143172e0418d5d0a7111822149b4919/src/liballoc/slice.rs#L134-L138). I'm still not fully getting why this is necessary, but let's accept it for now.